### DISCUSSION TITLE
Is it possible to get the Parent of an arbitrary Entity?

### DISCUSSION THREAD

**Discussion by Newbytee** [bevyengine/bevy#9292] [Q&A] [answered]
I know I can query an `Entity` along with its `Parent` from a system like so:

```rust
fn example_system(
    pairs: Query<(Entity, &Parent), (With<...>, Without<...>)>,
) {
    // Function body
}
```

But what if I want to get the `Parent` of an arbitrary `Entity` from within a function body? Is that possible? Does it even make sense? I would like to do it for debugging purposes, but I am not very familiar with Bevy in general, so I don't know if what I'm asking actually makes sense within Bevy's design. Still, if it is possible, I would like to know how I could achieve it.

**Comment by rparrett** [comment, marked as answer]
It is common and performant to use queries to retrieve components for a single `Entity` with [`Query::get`](https://docs.rs/bevy/latest/bevy/prelude/struct.Query.html#method.get).

```rust
fn example_system(parent_query: Query<&Parent>) {
    if let Ok(parent) = parent_query.get(entity) {
        // ...
    }
}
```

If you haven't already, check out [bevy-inspector-egui](https://github.com/jakobhellermann/bevy-inspector-egui) which is an extremely handy plugin that allows you to inspect your entities with a GUI.
